lib/kubernetes-deploy/deploy_task.rb

```diff
@@ -541,6 +543,29 @@ def confirm_namespace_exists
       @logger.info("Namespace #{@namespace} found")
     end
 
+    # make sure to never prune the ejson-keys secret
+    def check_applied_ejson_keys_secret
+      raw_secret, st, err = nil
+      with_retries(2) do
+        raw_secret, err, st = kubectl.run("get", "-n", @namespace, "secret",
+          "ejson-keys", "-oyaml", output_is_sensitive: true)
```

Comment:
Even if this code isn't in the provisioner, we should probably reuse its `EjsonSecretProvisioner::EJSON_KEYS_SECRET` constant